Generated code for HTTP basic authentication with ky does not compile

### Description

When using therefore with an Openapi schema that specifies HTTP basic authentication as the endpoint security method as bellow and using ky as the client, the generated code does not compile.

**Schema**
```
"security": [
    {
        "basic": []
    }
]
```

**Generated code**
```
if (basic !== undefined) {
    const [username, password] = typeof basic === 'function' ? await basic() : basic
    options.username = username
    options.password = password
}
```

**Compile errors**
```
error TS2339: Property 'username' does not exist on type 'KyRequest'.
error TS2339: Property 'password' does not exist on type 'KyRequest'
```

### Solution

I believe ky does not have an explicit way of providing the credentials, but for both ky and got they can be added as a header.

--- a/src/lib/primitives/restclient/security.ts
+++ b/src/lib/primitives/restclient/security.ts
@@ -47,8 +47,12 @@ export const toSecurityHook: {
                     .writeLine(`if (${name} !== undefined)`)
                     .block(() => {
                         hook.writeLine(`const [username, password] = typeof ${name} === 'function' ? await ${name}() : ${name}`)
-                            .writeLine('options.username = username')
-                            .writeLine('options.password = password')
+                            .conditionalWriteLine(client === 'got', 'options.username = username')
+                            .conditionalWriteLine(client === 'got', 'options.password = password')
+                            .conditionalWriteLine(
+                                client === 'ky',
+                                'options.headers.set("Authorization", `Basic ${btoa(`${username}:${password}`)}`)',
+                            )
                     })
             } else if (s.scheme === 'bearer') {
                 hook.writeLine(`const ${name} = this.auth.${name}!`)

